### PR TITLE
Add missing test coverage for edge cases, generics, and comparers

### PR DESCRIPTION
**Pull request by jonathanpeppers**
The existing test suite thoroughly covers the core sorting paths for all primitive types, but several API surfaces and input patterns were only tested for `int`. This PR fills those gaps, bringing the test count from 82 to 136.

## What's added

- **Edge-case patterns for more types** -- Already-sorted, reverse-sorted, all-duplicates, and duplicate-heavy tests now cover `byte`, `float`, `double`, and `string` (previously only `int`).
- **Generic `Sort<T>(T[], IComparer<T>)` overload** -- Correctness tests at sizes 27, 28, and 100 (fallback), plus null-array and null-comparer guard tests.
- **Span+comparer for more types** -- `Sort(Span<T>, IComparer<T>)` was only tested for `int`; now also covers `byte`, `double`, `long`, and `string`.
- **Reverse comparer tests** -- Verifies that custom comparers are actually applied (not silently ignored) across array, span, string, and generic overloads.
- **`nint`/`nuint` null-comparer defaults** -- These two types were missing from the existing `Sort_NullComparer_UsesDefault_Primitives` coverage.

All 136 tests pass.

### CODE REVIEW

File: SortingNetworks.Tests/NetworkSortTests.cs [hunk 1]
```diff
@@ -542,5 +542,420 @@ public void Sort_FallsBackForStrings()
...
+    [Fact]
+    public void Sort_Generic_RandomInts_MatchesArraySort()
+    {
+        var rng = new Random(42);
+        var input = Enumerable.Range(0, 27).Select(_ => rng.Next(-1000, 1000)).ToArray();
+        var expected = (int[])input.Clone();
+        Array.Sort(expected);
+
+        NetworkSort.Sort(input, Comparer<int>.Default);
+
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply SortingNetworks.Tests/NetworkSortTests.cs:760]: Fixed -- all generic tests now use explicit type arguments (e.g. `NetworkSort.Sort<int>(...)`) to ensure the generic overload is exercised. Also renamed this test to `Sort_Generic_27Elements_Int` for consistency.

File: SortingNetworks.Tests/NetworkSortTests.cs [hunk 2]
```diff
@@ -542,5 +542,420 @@ public void Sort_FallsBackForStrings()
...
+        Assert.Equal(expected, input);
+    }
+
+    [Fact]
+    public void Sort_Generic_NullArray_Throws()
+    {
+        Assert.Throws<ArgumentNullException>(() => NetworkSort.Sort((int[])null!, Comparer<int>.Default));
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply SortingNetworks.Tests/NetworkSortTests.cs]: Fixed -- now calls `NetworkSort.Sort<int>((int[])null!, Comparer<int>.Default)`.

File: SortingNetworks.Tests/NetworkSortTests.cs [hunk 3]
```diff
@@ -542,5 +542,420 @@ public void Sort_FallsBackForStrings()
...
+
+    [Fact]
+    public void Sort_Generic_Strings_WithOrdinalComparer()
+    {
+        var rng = new Random(42);
+        var input = Enumerable.Range(0, 27).Select(_ => rng.Next(0, 10000).ToString()).ToArray();
+        var expected = (string[])input.Clone();
+        Array.Sort(expected, StringComparer.Ordinal);
+
+        NetworkSort.Sort(input, StringComparer.Ordinal);
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply SortingNetworks.Tests/NetworkSortTests.cs]: Fixed -- now calls `NetworkSort.Sort<string>(input, StringComparer.Ordinal)`.

File: SortingNetworks.Tests/NetworkSortTests.cs [hunk 4]
```diff
@@ -542,5 +542,420 @@ public void Sort_FallsBackForStrings()
...
+        Array.Sort(expected);
+
+        NetworkSort.Sort(input, Comparer<int>.Default);
+
+        Assert.Equal(expected, input);
+    }
+
+    [Fact]
+    public void Sort_Generic_28Elements()
+    {
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply SortingNetworks.Tests/NetworkSortTests.cs]: Renamed to `Sort_Generic_28Elements_Int` to match the `Sort_<Size>Elements_<Type>` pattern.

File: SortingNetworks.Tests/NetworkSortTests.cs [hunk 5]
```diff
@@ -542,5 +542,420 @@ public void Sort_FallsBackForStrings()
...
+    }
+
+    [Fact]
+    public void Sort_Generic_NullComparer_Throws()
+    {
+        var array = new[] { "c", "a", "b" };
+        Assert.Throws<ArgumentNullException>(() => NetworkSort.Sort<string>(array, (IComparer<string>)null!));
+    }
+
+    [Fact]
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply SortingNetworks.Tests/NetworkSortTests.cs]: Removed -- this was a duplicate of `Sort_NullComparer_Throws_Generic`.

File: SortingNetworks.Tests/NetworkSortTests.cs [hunk 6]
```diff
@@ -542,5 +542,420 @@ public void Sort_FallsBackForStrings()
...
+    public void Sort_Generic_FallsBackForLargeArrays()
+    {
+        var rng = new Random(42);
+        var input = Enumerable.Range(0, 100).Select(_ => rng.Next()).ToArray();
+        var expected = (int[])input.Clone();
+        Array.Sort(expected);
+
+        NetworkSort.Sort(input, Comparer<int>.Default);
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply SortingNetworks.Tests/NetworkSortTests.cs]: Fixed -- now calls `NetworkSort.Sort<int>(input, Comparer<int>.Default)`.

File: SortingNetworks.Tests/NetworkSortTests.cs [hunk 7]
```diff
@@ -542,5 +542,420 @@ public void Sort_FallsBackForStrings()
...
+
+    [Fact]
+    public void Sort_Generic_28Elements()
+    {
+        var rng = new Random(42);
+        var input = Enumerable.Range(0, 28).Select(_ => rng.Next(-1000, 1000)).ToArray();
+        var expected = (int[])input.Clone();
+        Array.Sort(expected);
+
+        NetworkSort.Sort(input, Comparer<int>.Default);
```
[the remark this answers is not in the document]
Comment (jonathanpeppers) [reply SortingNetworks.Tests/NetworkSortTests.cs]: Fixed -- now calls `NetworkSort.Sort<int>(input, Comparer<int>.Default)`.
